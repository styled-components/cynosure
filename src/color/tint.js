// @flow
import curry from '../internalHelpers/_curry'
import mix from './mix'

/**
 * Tints a color by mixing it with white. `tint` can produce
 * hue shifts, where as `lighten` manipulates the luminance channel and therefore
 * doesn't produce hue shifts.
 *
 * @example
 * // Styles as object usage
 * const styles = {
 *   background: tint(0.25, '#00f')
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   background: ${tint(0.25, '#00f')};
 * `
 *
 * // CSS in JS Output
 *
 * element {
 *   background: "#bfbfff";
 * }
 */

function tint(percentage: number | string, color: string): string {
  return mix(parseFloat(percentage), color, 'rgb(255, 255, 255)')
}

const curriedTint = curry(tint)
export default curriedTint
