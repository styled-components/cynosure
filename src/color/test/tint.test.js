// @flow
import tint from '../tint'

describe('test', () => {
  it('should tint the provided color with white by the given percentage', () => {
    expect(tint(0.25, '#00f')).toMatchSnapshot()
  })
  it('should tint the provided color when passed a string for amount', () => {
    expect(tint('0.25', '#00f')).toMatchSnapshot()
  })
})
