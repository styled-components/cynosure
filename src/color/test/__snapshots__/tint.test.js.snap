// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should tint the provided color when passed a string for amount 1`] = `"#bfbfff"`;

exports[`test should tint the provided color with white by the given percentage 1`] = `"#bfbfff"`;
